contracts/_external/BytesLib.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: Unlicensed
```

Comment:
Is this audited or used in production anywhere?
Same with StringUtils

Reply:
The 2 external libraries
* [solidity-bytes-utils](https://github.com/GNSPS/solidity-bytes-utils) is used by Autherium, Optimism and a few others.

A recent audit found a minor issue with the `slice` method (which we use). But we should be using v0.1.2 which fixes this (just updated it)..

https://github.com/GNSPS/solidity-bytes-utils/blob/master/README.md

```
A bug regarding zero-length slices was disclosed by @MrChico following an audit to the Optimism codebase.

The exact bug happened under the following conditions: if memory slots higher then the current free-memory pointer
 were tainted before calling the slice method with a desired length of 0, the returned bytes array, instead of being a 
zero-length slice was an array of arbitrary length based on the values that previously populated that memory region.

Overall, the usage of zero-length slices should be pretty unusual and, as such, hopefully, this bug does not have far-
reaching implications. Nonetheless, please update the library to the new version if you're using it in production.

TL;DR: if you're using the slice method with a length parameter of '0' in your codebase, please update to version 0.1.2
 of the bytes library ASAP!
```

* There's aren't a widely used library for String utilities in solidity. The most used one is [Arachnid/solidity-stringutils](https://github.com/Arachnid/solidity-stringutils/blob/master/src/strings.sol). We could use that.

However since we dont require extensive string operations just a function to convert `uint` to `bytes/string`, I used [pipermerriam/ethereum-string-utils](https://github.com/pipermerriam/ethereum-string-utils/blob/master/contracts/StringLib.sol).

I dont think either of these libraries are audited we should do the due diligence ourselves.

***

The first library is not very critical as its invoked only when the transaction reverts. 
The 2nd one is probably a little more higher risk, but it's a very small function which I assume all of us can manuall verify.